File: Neocom/Neocom/UIColor+NC.m
//
//  UIColor+NC.m
//  Neocom
//
//  Created by Artem Shimanski on 15.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

#import "UIColor+NC.h"

@implementation UIColor (NC)

+ (instancetype) colorWithString:(NSString*) string {
	float rgba[] = {0,0,0,1};
	
	NSRange range = NSMakeRange(0, string.length);
	if ([string characterAtIndex:0] == '#') {
		range.length--;
		range.location++;
	}
	int i = 0;
	while (range.length > 0) {
		NSUInteger len = MIN(2, range.length);
		NSScanner* scanner = [NSScanner scannerWithString:[string substringWithRange:NSMakeRange(range.location, len)]];
		range.location += len;
		range.length -= len;
		unsigned int n = 0;
		if ([scanner scanHexInt:&n])
			rgba[i++] = n / 255.0;
		else
			return nil;
	}
	return [UIColor colorWithRed:rgba[0] green:rgba[1] blue:rgba[2] alpha:rgba[3]];
}

+ (instancetype) colorWithUInteger:(NSUInteger) value {
	const Byte* abgr = (const Byte*) &value;
	return [UIColor colorWithRed:abgr[3] / 255.0 green:abgr[2] / 255.0 blue:abgr[1] / 255.0 alpha:abgr[0] / 255.0];
}

+ (instancetype) backgroundColor {
	return [UIColor colorWithUInteger:0x172126FF];
}


@end

File: Neocom/Neocom/UIColor+NC.swift
//
//  UIColor+NC.swift
//  Neocom
//
//  Created by Artem Shimanski on 14.12.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

extension UIColor {
	convenience init(security: Float) {
		if security >= 1.0 {
			self.init(number: 0x2FEFEFFF)
		}
		else if security >= 0.9 {
			self.init(number: 0x48F0C0FF)
		}
		else if security >= 0.8 {
			self.init(number: 0x00EF47FF)
		}
		else if security >= 0.7 {
			self.init(number: 0x00F000FF)
		}
		else if security >= 0.6 {
			self.init(number: 0x8FEF2FFF)
		}
		else if security >= 0.5 {
			self.init(number: 0xEFEF00FF)
		}
		else if security >= 0.4 {
			self.init(number: 0xD77700FF)
		}
		else if security >= 0.3 {
			self.init(number: 0xF06000FF)
		}
		else if security >= 0.2 {
			self.init(number: 0xF04800FF)
		}
		else if security >= 0.1 {
			self.init(number: 0xD73000FF)
		}
		else {
			self.init(number: 0xF00000FF)
		}
	}
	
}
